### DISCUSSION TITLE
Problema actualizando subpropiedades a través de un bot

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#61] [2. Questions and Answers] [answered]
Hola, tengo una consulta:

Resulta que tengo un formulario con 3 selectores de propiedades, donde puedo seleccionar mas de 1 elemento en cada uno de ellos. 

Lo que quiero hacer es que mi asset se agreguen como subpropiedades cada uno de estos elementos, ya que con mi código adjunto no me está funcionando el update property y creo que tengo malo las rutinas de updateproperty
Adjunto el código completo. 

Los titulos son: 
*  change state - 
*  NWR asset - 
*  iterare 1 - 
*  update asset 1 - 
*  iterate 2 - 
*  update asset 2 - 
*  iterate 3 - 
*  update asset 3 - 
*  send message

```
surveyTriggers" : [
        {
            "_id" : ObjectId("606f79c843bb92000972a8a5"), 
            "surveyId" : ObjectId("606f4efb50736600095facdb"), 
            "triggers" : [
                {
                    "_id" : ObjectId("606f79c843bb92000972a8a6"), 
                    "start" : "get_asset", 
                    "maxIterations" : NumberInt(100), 
                    "stages" : [
                        {
                            "_id" : ObjectId("606f6e7950736600095ff37b"), 
                            "key" : "change_state", 
                            "name" : "PBChangeState", 
                            "data" : {
                                "tid" : "$VALUE#_id", 
                                "smState" : "6054f6d67c8ec9000904bf81", 
                                "taskGroup" : "6054f4e16aca6000088a355e"
                            }, 
                            "next" : {
                                "DEFAULT" : "iterate_imagen"
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("605cab99f2882500085a6fde"), 
                            "key" : "send_ack", 
                            "name" : "PBMessage", 
                            "data" : {
                                "content" : "Se han asignado las áreas respectivas para realizar las acciones del paciente", 
                                "contentType" : "text/system", 
                                "sentBy" : "$VALUE#user", 
                                "channelIds" : "$VALUE#channel"
                            }, 
                            "next" : {
                                "DEFAULT" : ""
                            }, 
                            "customNetworkRequest" : [                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f6e7950736600095ff37d"), 
                            "key" : "iterate_imagen", 
                            "name" : "FCEach", 
                            "data" : {
                                "control" : "$VALUE#sentAnswer|data|[find=>identifier=acciones_imagenes_hepatopancreaticobiliar]|process", 
                                "target" : "imagen"
                            }, 
                            "next" : {
                                "STEP" : "update_imagen", 
                                "DONE" : "iterate_procedimientos"
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f79c843bb92000972a8aa"), 
                            "key" : "update_imagen", 
                            "name" : "PBUpdateProperty", 
                            "data" : {
                                "propertyId" : "$OUTPUT#get_asset#data|_id", 
                                "display" : "$OUTPUT#get_asset#data|name|display", 
                                "code" : "$OUTPUT#get_asset#data|name|code", 
                                "propertyType" : "$OUTPUT#get_asset#data|propertyType", 
                                "subproperty" : [
                                    [
                                        "$OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#imagen]"
                                    ]
                                ]
                            }, 
                            "next" : {
                                "DEFAULT" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f6e7950736600095ff37f"), 
                            "key" : "iterate_procedimientos", 
                            "name" : "FCEach", 
                            "data" : {
                                "control" : "$VALUE#sentAnswer|data|[find=>identifier=acciones_procedimientos_hepatopancreaticobiliar]|process", 
                                "target" : "procedimientos"
                            }, 
                            "next" : {
                                "STEP" : "update_procedimientos", 
                                "DONE" : "iterate_otros"
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f79c843bb92000972a8ac"), 
                            "key" : "update_procedimientos", 
                            "name" : "PBUpdateProperty", 
                            "data" : {
                                "propertyId" : "$OUTPUT#get_asset#data|_id", 
                                "display" : "$OUTPUT#get_asset#data|name|display", 
                                "code" : "$OUTPUT#get_asset#data|name|code", 
                                "propertyType" : "$OUTPUT#get_asset#data|propertyType", 
                                "subproperty" : [
                                    [
                                        "$OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#procedimientos]"
                                    ]
                                ]
                            }, 
                            "next" : {
                                "DEFAULT" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f767443bb920009729ad2"), 
                            "key" : "iterate_otros", 
                            "name" : "FCEach", 
                            "data" : {
                                "control" : "$VALUE#sentAnswer|data|[find=>identifier=acciones_otras_hepatopancreaticobiliar]|process", 
                                "target" : "otros"
                            }, 
                            "next" : {
                                "STEP" : "update_otros", 
                                "DONE" : "send_ack"
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f79c843bb92000972a8ae"), 
                            "key" : "update_otros", 
                            "name" : "PBUpdateProperty", 
                            "data" : {
                                "propertyId" : "$OUTPUT#get_asset#data|_id", 
                                "display" : "$OUTPUT#get_asset#data|name|display", 
                                "code" : "$OUTPUT#get_asset#data|name|code", 
                                "propertyType" : "$OUTPUT#get_asset#data|propertyType", 
                                "subproperty" : [
                                    [
                                        "$OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#otros]"
                                    ]
                                ]
                            }, 
                            "next" : {
                                "DEFAULT" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }, 
                        {
                            "_id" : ObjectId("606f75d7507366000960067a"), 
                            "key" : "get_asset", 
                            "name" : "NWRequest", 
                            "data" : {
                                "url" : "$JOIN#/#($ENV#BASEURL)#(api)#(properties)#($VALUE#asset)", 
                                "method" : "GET", 
                                "defaultAuth" : true
                            }, 
                            "next" : {
                                "SUCCESS" : "change_state", 
                                "ERROR" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }
                    ], 
                    "version" : "v3"
                }
            ]
        }, 
]
``` 

gracias

**Comment by acedward** [comment, marked as answer]
@NicolasPLG Esto me parece extraño, está bien el doble `[ [ ] ]`
`"subproperty" : [["$OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#imagen]"]]`

Además `[add=>string=a]` retorna un Arreglo, entonces tiene un triple-arreglo en este caso

**Reply by NicolasPLG** [reply]
cuando lo coloco en la rutina me pide agregar un [""] para que el código funcione.
Visualmente se ve asi
![image](https://user-images.githubusercontent.com/61419471/114102801-fd2bfe80-9895-11eb-8b5e-34ac27ba66f3.png)
y si quito los [""] me arroja que no es código de Json
![image](https://user-images.githubusercontent.com/61419471/114102849-15038280-9896-11eb-96dc-49b77dfe4a0e.png)

Quizas debo colocarlo en modo avanzado?

**Reply by acedward** [reply]
Si, puedes dejar el mismo string, pero en avanzado

**Reply by NicolasPLG** [reply]
Me arrojó  en la consola. Probaré con quitar los [] en la base. El bot tiene *-write
ERROR: 18838 Error: ERROR: (18838) [500] {
  "body": {
    "error": "ValidationError: subproperty: Cast to Array failed for value \"[ [ '606f51f143bb92000972497f', '606f515a50736600095fb267' ] ]\" at path \"subproperty\""
  },

**Reply by NicolasPLG** [reply]
Me funcionó me agrega 1 y no todos. Quité el [] desde la base dejando asi  "subproperty" : "$OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#procedimientos]" y no tuve problemas. Levantaré la incidencia para que lo puedan ver

Quizas deba usar un nwr de metodo Patch en vez del update propery. lo revisaré gracias @acedward por tu ayuda

**Reply by acedward** [reply]
¿Entiendo que está todo funcionando ahora?

Lo que está pasando es que  el campo  "subproperty"  recibe un arreglo de ids y  `$OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#procedimientos]` retorna un arreglo,  al agregar otro `[ ]`, estas generando un arreglo de arreglos causando el comportamiento no esperado.

**Reply by NicolasPLG** [reply]
@acedward lo revisé recién en mi flujo y no me funciona :(.
¿Cómo lo puedo hacer? me puedes sugerir un cambio de bot o cambio en la rutina del update property?

**Reply by acedward** [reply]
puedes setear `debug: true` aquí:
```
triggers" : [
                {
                    "_id" : ObjectId("606f79c843bb92000972a8a6"), 
                   "debug": true,
                    "start" : "get_asset", 
                    "maxIterations" : NumberInt(100), 
                    "stages" : [
```
Me avisas cuando lo ejecutes y puedo mirar los logs

**Reply by NicolasPLG** [reply]
@acedward eso arroja los logs del reporte no sé si está bien. Entré a la librería shedulers
´´´
{ 
    "_id" : ObjectId("60759ea1ed27578d2ab7c6d9"), 
    "op" : "failed", 
    "diff" : [

    ], 
    "details" : "{\"err\":\"{\\n  \\\"error\\\": \\\"Error: $OUTPUT#get_asset#data|subproperty|[add=>string=$VALUE#imagen]\\\\n↑                                 ↑\\\\n┴─────────────────────────────────┘\\\\n\\\\nUnexpected end of input: \\\\n|[add=>string=$VALUE#imagen]\\\",\\n  \\\"data\\\": {\\n    \\\"contextNames\\\": [\\n      \\\"iterate_imagen::0\\\"\\n    ]\\n  }\\n}\\nundefined\",\"output\":{\"debug\":{\"results\":[{\"key\":\"get_asset\",\"name\":\"NWRequest\",\"status\":\"SUCCESS\",\"result\":{\"statusCode\":200,\"data\":{\"_id\":\"6058f4ca6aca600008949854\",\"subproperty\":[\"606f534150736600095fb681\",\"606f513743bb920009724794\"],\"users\":[],\"isActive\":true,\"search\":[\"1\",\"9\",\"tester\",\"unique\",\"6058f4cafc663ab4e4cc9d0d\",\"19\",\"unique6058f4cafc663ab4e4cc9d0d\"],\"company\":\"5e0a1a1345e9602068206fc9\",\"propertyType\":\"cancer_hepatopancreaticobiliar_assets\",\"name\":{\"code\":\"unique-6058f4cafc663ab4e4cc9d0d\",\"display\":\"1-9 - tester - tester\"},\"createdAt\":\"2021-03-22T19:49:30.817Z\",\"modifiedAt\":\"2021-04-13T13:37:36.739Z\",\"__v\":20,\"schemaInstance\":{},\"extra\":{},\"superProperties\":[]}},\"nextBot\":\"change_state\",\"iteration\":1},{\"key\":\"change_state\",\"name\":\"PBChangeState\",\"status\":\"DEFAULT\",\"result\":{\"task\":{\"public\":{\"isActive\":false,\"language\":\"spanish\",\"title\":\"\",\"response\":\"\"},\"answers\":[\"6058f4b72cdb27088b96b08f\",\"605cac032cdb27088bdcc516\",\"605cad2b2cdb27088bf57a4f\",\"605cad4c2cdb27088bd4e785\",\"606f75672cdb27088baf42a6\",\"606f79652cdb27088be2a0fa\",\"606f81c42cdb27088b0d8508\",\"60759e402cdb27088beb0a07\"],\"channelType\":\"bound\",\"indentation\":0,\"isActive\":true,\"isValid\":false,\"activeSlas\":[],\"validators\":[],\"editors\":[\"5e0a2cdb7e7bc5207a27088b\"],\"followers\":[],\"visibility\":[],\"userList\":[\"5e0a2cdb7e7bc5207a27088b\"],\"color\":\"none\",\"formMessage\":[],\"_id\":\"6058f4cafc663ab4e4cc9d0d\",\"name\":\"1-9 - tester - tester\",\"relativeWeight\":1,\"assignee\":\"5e0a2cdb7e7bc5207a27088b\",\"startDate\":\"2021-03-22T12:00:00.000Z\",\"createdBy\":\"5e0a2cdb7e7bc5207a27088b\",\"company\":\"5e0a1a1345e9602068206fc9\",\"taskGroup\":\"6054f4e1f28825000841a279\",\"owner\":\"manual\",\"weight\":0,\"status\":\"6054f0bb7c8ec9000904863a\",\"smState\":\"6054f6d67c8ec9000904bf81\",\"smStateMachine\":\"6054f5ad9297b9000843bcd0\",\"modifiedStateAt\":\"2021-04-13T13:37:37.321Z\",\"closedAt\":null,\"asset\":\"6058f4ca6aca600008949854\",\"channel\":\"6058f4ca6aca600008949858\",\"projectCode\":\"6058f4ca6aca600008949859\",\"serial\":1,\"createdAt\":\"2021-03-22T19:49:30.954Z\",\"modifiedAt\":\"2021-04-13T13:37:37.607Z\",\"__v\":12,\"endDate\":\"2021-03-25T15:33:39.350Z\",\"estimatedTime\":null,\"status1\":null,\"status2\":null,\"status3\":null,\"status4\":null,\"status5\":null}},\"nextBot\":\"iterate_imagen\",\"iteration\":2}],\"input\":{\"__meta\":{\"_id\":\"60759ea089429dcf37f937c0\",\"code\":\"6058f4cafc663ab4e4cc9d0d#6054f6c16aca6000088a466e#1618321056853\"},\"public\":{\"isActive\":false,\"language\":\"spanish\",\"title\":\"\",\"response\":\"\"},\"answers\":[\"6058f4b72cdb27088b96b08f\",\"605cac032cdb27088bdcc516\",\"605cad2b2cdb27088bf57a4f\",\"605cad4c2cdb27088bd4e785\",\"606f75672cdb27088baf42a6\",\"606f79652cdb27088be2a0fa\",\"606f81c42cdb27088b0d8508\",\"60759e402cdb27088beb0a07\"],\"channelType\":\"bound\",\"indentation\":0,\"isActive\":true,\"isValid\":false,\"activeSlas\":[],\"validators\":[],\"editors\":[\"5e0a2cdb7e7bc5207a27088b\"],\"followers\":[],\"visibility\":[],\"userList\":[\"5e0a2cdb7e7bc5207a27088b\"],\"color\":\"none\",\"formMessage\":[],\"_id\":\"6058f4cafc663ab4e4cc9d0d\",\"name\":\"1-9 - tester - tester\",\"relativeWeight\":1,\"assignee\":\"5e0a2cdb7e7bc5207a27088b\",\"startDate\":\"2021-03-22T12:00:00.000Z\",\"createdBy\":\"5e0a2cdb7e7bc5207a27088b\",\"company\":\"5e0a1a1345e9602068206fc9\",\"taskGroup\":\"6054f4e1f28825000841a279\",\"owner\":\"manual\",\"weight\":0,\"status\":\"6054f08bf288250008417721\",\"smState\":\"6054f6c16aca6000088a466e\",\"smStateMachine\":\"6054f5ad9297b9000843bcd0\",\"modifiedStateAt\":\"2021-04-13T13:35:53.561Z\",\"closedAt\":null,\"asset\":\"6058f4ca6aca600008949854\",\"channel\":\"6058f4ca6aca600008949858\",\"projectCode\":\"6058f4ca6aca600008949859\",\"serial\":1,\"createdAt\":\"2021-03-22T19:49:30.954Z\",\"modifiedAt\":\"2021-04-13T13:37:36.817Z\",\"__v\":12,\"endDate\":\"2021-03-25T15:33:39.350Z\",\"estimatedTime\":null,\"status1\":null,\"status2\":null,\"status3\":null,\"status4\":null,\"status5\":null,\"sentAnswer\":{\"score\":{\"scores\":[]},\"isAuto\":true,\"properties\":[\"6054f08bf288250008417721\",\"6058f4ca6aca600008949854\",\"606f515a50736600095fb267\",\"606f514943bb9200097247e2\"],\"propertyTypes\":[\"cancer_hepatopancreaticobiliar_sm\",\"cancer_hepatopancreaticobiliar_assets\",\"imagenes_hepatopancreaticobillar\"],\"extendsAnswer\":[],\"rExtendsAnswer\":[],\"_id\":\"60759ea036b40e00093a582b\",\"data\":[{\"code\":[],\"display\":[\"## Ingrese acciones a realizar\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a582c\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f4efc50736600095facdc\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"asignacion_area_hepatopancreaticobiliar\",\"diff\":[]},{\"code\":[],\"display\":[\"## Imagenes\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a582d\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee03d\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"imagenes_area_hepatopancreaticobiliar\",\"diff\":[]},{\"code\":[],\"display\":[\"Ingrese acciones imagenes\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a582e\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee03f\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"acciones_imagenes_hepatopancreaticobiliar_1617908895907\",\"diff\":[]},{\"code\":[\"{\\\"allow\\\":\\\"propertyType\\\",\\\"value\\\":\\\"imagenes_hepatopancreaticobillar\\\",\\\"filter\\\":\\\"*\\\"}\"],\"display\":[],\"responses\":[\"{\\\"_id\\\":\\\"606f515a50736600095fb267\\\",\\\"subproperty\\\":[],\\\"users\\\":[],\\\"isActive\\\":true,\\\"name\\\":{\\\"code\\\":\\\"biopsia_bajo_rayos_imagenes\\\",\\\"display\\\":\\\"Biopsia bajo rayos\\\"},\\\"propertyType\\\":\\\"imagenes_hepatopancreaticobillar\\\",\\\"company\\\":\\\"5e0a1a1345e9602068206fc9\\\",\\\"createdAt\\\":\\\"2021-04-08T18:54:18.343Z\\\",\\\"modifiedAt\\\":\\\"2021-04-08T18:54:18.374Z\\\",\\\"__v\\\":0,\\\"color\\\":\\\"#F4A610\\\"}\",\"{\\\"_id\\\":\\\"606f514943bb9200097247e2\\\",\\\"subproperty\\\":[],\\\"users\\\":[],\\\"isActive\\\":true,\\\"name\\\":{\\\"code\\\":\\\"colangio_resonancia_imagenes\\\",\\\"display\\\":\\\"Colangio resonancia\\\"},\\\"propertyType\\\":\\\"imagenes_hepatopancreaticobillar\\\",\\\"company\\\":\\\"5e0a1a1345e9602068206fc9\\\",\\\"createdAt\\\":\\\"2021-04-08T18:54:01.303Z\\\",\\\"modifiedAt\\\":\\\"2021-04-08T18:54:01.333Z\\\",\\\"__v\\\":0,\\\"color\\\":\\\"#4210E0\\\"}\"],\"process\":[\"606f515a50736600095fb267\",\"606f514943bb9200097247e2\"],\"_id\":\"60759ea036b40e00093a582f\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee042\",\"contentType\":\"application/vnd.cotalker.survey+property\",\"identifier\":\"acciones_imagenes_hepatopancreaticobiliar\",\"diff\":[]},{\"code\":[],\"display\":[\"## Procedimientos\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a5830\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee044\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"procedimientos_area_hepatopancreaticobiliar\",\"diff\":[]},{\"code\":[],\"display\":[\"Ingrese acciones procedimientos\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a5831\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee046\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"acciones_procedimientos_hepatopancreaticobiliar_1617908895907\",\"diff\":[]},{\"code\":[\"{\\\"allow\\\":\\\"propertyType\\\",\\\"value\\\":\\\"procedimientos_hepatopancreaticobillar\\\",\\\"filter\\\":\\\"*\\\"}\"],\"display\":[],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a5832\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee049\",\"contentType\":\"application/vnd.cotalker.survey+property\",\"identifier\":\"acciones_procedimientos_hepatopancreaticobiliar\",\"diff\":[]},{\"code\":[],\"display\":[\"## Otras especialidades\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a5833\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee04b\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"otras_area_hepatopancreaticobiliar\",\"diff\":[]},{\"code\":[],\"display\":[\"Ingrese acciones otras especialidades\"],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a5834\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee04d\",\"contentType\":\"application/vnd.cotalker.survey+text\",\"identifier\":\"acciones_otras_hepatopancreaticobiliar_1617908895907\",\"diff\":[]},{\"code\":[\"{\\\"allow\\\":\\\"propertyType\\\",\\\"value\\\":\\\"otras_especialidades_hepatopancreaticobillar\\\",\\\"filter\\\":\\\"*\\\"}\"],\"display\":[],\"responses\":[],\"process\":[],\"_id\":\"60759ea036b40e00093a5835\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"question\":\"606f54a18f50df0009aee050\",\"contentType\":\"application/vnd.cotalker.survey+property\",\"identifier\":\"acciones_otras_hepatopancreaticobiliar\",\"diff\":[]}],\"uuid\":\"60759e402cdb27088beb0a07\",\"survey\":\"606f4efb50736600095facdb\",\"company\":\"5e0a1a1345e9602068206fc9\",\"user\":\"5e0a2cdb7e7bc5207a27088b\",\"createdAt\":\"2021-04-13T13:37:36.500Z\",\"startDate\":\"2021-04-13T13:37:36.500Z\",\"modifiedAt\":\"2021-04-13T13:37:36.519Z\",\"target\":null,\"channel\":\"6058f4ca6aca600008949858\",\"__v\":0}}}}}", 
    "scheduleId" : ObjectId("60759ea089429dcf37f937c0"), 
    "company" : ObjectId("5e0a1a1345e9602068206fc9"), 
    "createdAt" : ISODate("2021-04-13T13:37:37.777+0000"), 
    "__v" : NumberInt(0)
}
´´´

**Reply by acedward** [reply]
Súper, 

Cuando aparece esto en los logs:
```
↑
┴─────────────────────────────────┘
```
Es un error de COTLang, significa que no hay un error em la sintaxis.

Esto es inválido `[add=>string=$VALUE#imagen]`
Falta un paréntesis: `[add=>string=($VALUE#imagen)]`

La documentación oficial del lenguaje está en: 
https://www.cotalker.com/function/cotlang/

**Reply by NicolasPLG** [reply]
Gracias @acedward lo probaré y revisaré la documentación en esa parte. Pensé inicialmente que no debería tener ya que cuando se ocupa la estructura $VALUE#sentAnswer|data|[find=>identifier="*"]|process, no lleva paréntesis, pero ahora me di cuenta que lo otro es distinto porque proviene de un iterar en vez de una respuesta directa al survey.

Gracias
